### DISCUSSION TITLE
find_tables() 'cells' attributes format

### DISCUSSION THREAD

**Discussion by isaac-peter** [pymupdf/PyMuPDF#3629] [Looking for help] [answered]
Hello,

The extract() method of find_tables() seems to extract using the following format: Each element is a table row, and each subelement is a column in that given row.

The cells attribute on the other hand produces a list where each element is a cell, and the subelement is the bbox. 

How do I relate a given element in the cells attribute back to the cell in the extract() method? Said otherwise, how do I find the bbox of a given cell from the extract() method?

I don't believe the "Page" doc describes the structure of the extract() or cells attribute.

thanks!

**Comment by isaac-peter** [comment]
After reviewing the doc again, I'm thinking that maybe I use the row attribute to get the cells bboxes, and then relate this back to the extract() structure I described above?

**Comment by JorjMcKie** [comment, marked as answer]
This has been answered in #3587. E.g.

```python
imglist = page.get_image_info()

# copy of the table's text content:
tab_text = tab.extract()[:]
# the table's cell bboxes as Rect objects:
tab_cells=[[pymupdf.Rect(c) for c in r.cells] for r in tab.rows]
```
Are 2 lists of lists with the same sizes and indexed as [row][col]. So the text in `tab_text[row][col]` has the cell coordinates `tab_cells[row][col]` (which is a `Rect` object).

**Reply by isaac-peter** [reply]
Thanks!
